### PR TITLE
IGNITE-18556: Sql. TypeSystem. Default implementation of getDefaultPrecision for FLOAT and DOUBLE returns the same value.

### PR DESCRIPTION
**Pull request by lowka**
Fixes precision for `FLOAT` data type and adds tests for `IgniteTypeSystem` and removes `Serializable` marker from it.

https://issues.apache.org/jira/browse/IGNITE-18556

---

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystemTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        assertEquals(0, typeSystem.getMaxPrecision(SqlTypeName.DATE));
+        assertEquals(TIME_PRECISION, typeSystem.getMaxPrecision(SqlTypeName.TIME));
+        assertEquals(TIME_PRECISION, typeSystem.getMaxPrecision(SqlTypeName.TIME_WITH_LOCAL_TIME_ZONE));
+        assertEquals(TIME_PRECISION, typeSystem.getMaxPrecision(SqlTypeName.TIMESTAMP));
+        assertEquals(TIME_PRECISION, typeSystem.getMaxPrecision(SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE));
+    }
+
+    @Test
+    public void getDefaultPrecision() {
+        assertEquals(STRING_PRECISION, typeSystem.getMaxPrecision(SqlTypeName.CHAR));
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystemTest.java:60]: getDefaultPrecision

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystem.java [hunk 2]
```diff
@@ -61,19 +61,19 @@ public int getMaxPrecision(SqlTypeName typeName) {
...
                 return 6;
-            // FLOAT is an alias for REAL, When choosing between a real and a float, let's always choose a real.
-            // getDefaultPrecision for types that do not have precision only affects results of TypeFactory::leastRestrictive
-            case REAL:
-                return 8;
             case FLOAT:
-                return 7;
+                // Although FLOAT is an alias for REAL, we cannot use the same precision for them, w/o making
+                // results of TypeFactory::LeastRestrictiveType() non-deterministic. 
+                // We need to change the FLOAT precision, because by default calcite uses the precision for FLOAT and DOUBLE.
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystem.java:71]: do you mean 'the same'?

Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystem.java:71]: the same precision - equal precision

Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystem.java]: Fixed.

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**